src/amo/components/RatingsByStar/index.js

```diff
@@ -86,85 +90,104 @@ export class RatingsByStarBase extends React.Component<InternalProps> {
+        case '4':
+          return i18n.gettext(`Read all ${count} four-star reviews`);
+        case '3':
+          return i18n.gettext(`Read all ${count} three-star reviews`);
+        case '2':
+          return i18n.gettext(`Read all ${count} two-star reviews`);
+        case '1':
+          return i18n.gettext(`Read all ${count} one-star reviews`);
+        default:
+          return '';
+      }
     };
 
     return (
       <div className="RatingsByStar">
         {errorHandler.renderErrorIfPresent()}
-        <div className="RatingsByStar-graph">
-          {['5', '4', '3', '2', '1'].map((star) => {
-            let starCount;
-            let starCountNode;
-
-            function createLink(text) {
-              invariant(addon, 'addon was unexpectedly empty');
+        <table className="RatingsByStar-graph">
+          <tbody>
+            {['5', '4', '3', '2', '1'].map((star) => {
+              let starCount;
+              let starCountNode;
+
+              function createLink(body) {
+                invariant(addon, 'addon was unexpectedly empty');
+
+                return (
+                  <tr
+                    title={getLinkTitle(star, starCount) || ''}
+                    onClick={() => self.handleRouting(star)}
+                    className="RatingByStar-table-row"
+                  >
+                    {body}
+                  </tr>
+                );
+              }
 
-              return (
-                <Link
-                  title={linkTitles[star] || ''}
-                  to={reviewListURL({
-                    addonSlug: addon.slug,
-                    score: star,
-                    src: location.query.src,
-                  })}
-                >
-                  {text}
-                </Link>
-              );
-            }
+              if (!errorHandler.hasError()) {
+                if (groupedRatings) {
+                  starCount = groupedRatings[star];
+                }
 
-            if (!errorHandler.hasError()) {
-              if (groupedRatings) {
-                starCount = groupedRatings[star];
+                starCountNode = i18n.formatNumber(starCount || 0);
               }
 
-              starCountNode = loading ? (
-                <LoadingText minWidth={95} />
-              ) : (
-                createLink(i18n.formatNumber(starCount || 0))
-              );
-            }
-
-            return (
-              <React.Fragment key={star}>
-                <div className="RatingsByStar-star">
-                  {loading ? (
-                    <LoadingText minWidth={95} />
-                  ) : (
-                    createLink(i18n.formatNumber(star))
-                  )}
-                  <IconStar selected />
-                </div>
-                <div className="RatingsByStar-barContainer">
+              return (
+                <React.Fragment key={star}>
                   {loading ? (
```

Comment:
I wonder if it would be a bit more efficient and cleaner if you were to produce the contents of the `<tr>` via a function, which returns either the loading version or the non-loading version, and then here either wrap that in `createLink` or not, depending on `loading`? This would allow you to avoid all the duplicate code for each `th` tag.